### ISSUE TITLE
Stuck at phase 7 on installation wizzard

### ISSUE THREAD

**Issue by sfn84** [EarthLab-Luxembourg/docker-centreon#4]
Hi,

I first would like to thank you for your effort on working to that docker-centreon project.

I am trying to work through this docker-compose file to facilitate my deployment:
[docker-compose.yml.txt](https://github.com/EarthLab-Luxembourg/docker-centreon/files/4465337/docker-compose.yml.txt)

Unfortunately, I am facing an issue in this step:
![image](https://user-images.githubusercontent.com/1378609/79053826-8f2c0780-7c40-11ea-8042-3b4901cf1c18.png)

It looks stucked on "Setting up basic configuration" and loop forever...
When I reboot the container, it stays stuck.
Have you ever got this problem?

I tried using other mariadb instance (in the host) and I got the same issue.

Some information about my mariadb instance while it's freezing:
Server version: 10.1.44-MariaDB-0+deb9u1 Debian 9.11
MariaDB [(none)]> show databases;
+--------------------+
| Database           
+--------------------+
| centreon           
| centreon_storage   
| information_schema 
| mysql              
| performance_schema 
+--------------------+

Thank you

**Comment by eLvErDe** [comment]
Hi,

You should spawn a shell in the container and check /var/log/centreon and /var/log/apache2, you'll probably find something related to the error

**Comment by sfn84** [comment]
Hi,

I got this error in /var/log/centreon/centreon-error.log:

```
[11-Apr-2020 21:54:59 UTC] PHP Fatal error:  Uncaught TypeError: Argument 1 passed to CentreonLegacy\Core\Configuration\Configuration::__construct() must be of the type array, null given, called in /srv/centreon/src/CentreonLegacy/ServiceProvider.php on line 104 and defined in /srv/centreon/src/CentreonLegacy/Core/Configuration/Configuration.php:70
Stack trace:
#0 /srv/centreon/src/CentreonLegacy/ServiceProvider.php(104): CentreonLegacy\Core\Configuration\Configuration->__construct(NULL, '/srv/centreon/', Object(Symfony\Component\Finder\Finder))
#1 /srv/centreon/vendor/pimple/pimple/src/Pimple/Container.php(118): CentreonLegacy\ServiceProvider->CentreonLegacy\{closure}(Object(Pimple\Container))
#2 /srv/centreon/vendor/pimple/pimple/src/Pimple/Psr11/ServiceLocator.php(65): Pimple\Container->offsetGet('configuration')
#3 /srv/centreon/src/CentreonLegacy/Core/Utils/Utils.php(142): Pimple\Psr11\ServiceLocator->get('configuration')
#4 /srv/centreon/src/CentreonLegacy/Core/Utils/Utils.php(105): CentreonLegacy\Core\Utils\Utils->replaceMacros('\nINSERT INTO `n. in /srv/centreon/src/CentreonLegacy/Core/Configuration/Configuration.php on line 70
```

I wonder if the form I filled in the previous steps are correct as I noticed I had to adapt/create some paths that did not exist..
I will see tomorrow if I find anything about this.

Thanks Adam.

**Comment by eLvErDe** [comment]
You're not supposed to adapt anything beyond MySQL params :D

**Comment by sfn84** [comment]
I did not :-)
I meant here:
![image](https://user-images.githubusercontent.com/1378609/79065173-883bde00-7cae-11ea-85d9-3235d3fa5016.png)
and here:
![image](https://user-images.githubusercontent.com/1378609/79065205-dcdf5900-7cae-11ea-87f4-3efdd092e9e9.png)

For the db I used this for the mariadb container:
![image](https://user-images.githubusercontent.com/1378609/79065277-7444ac00-7caf-11ea-896d-e4811907e60b.png)

and tried with my mariadb host instance that way:
![image](https://user-images.githubusercontent.com/1378609/79065315-b2da6680-7caf-11ea-9564-fe46dc6698ad.png)

and before going further I did this in mysql instance as mentioned in your README:
`CREATE USER 'root'@'%' IDENTIFIED BY 'secret';
GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' WITH GRANT OPTION;
FLUSH PRIVILEGES;`

And in any case I got the same issue. It is kind of a bug as it creates well centreon and centreon_storage databases.
And it keeps looping indefinitely without any explicit error message except the one I sent above in the centreon-error.log file.

I used the latest DockerFile building my image.
This in CONFIG file:
```
hostname=centreon
host_port=8080
host_volumes_path=/srv/docker/centreon
container_name=centreon
smtp_server=172.17.0.1
vmware_perl_sdk="VMware-vSphere-Perl-SDK-6.0.0-3561779.x86_64.tar.gz"
```

**Comment by eLvErDe** [comment]
Can you stop/rm the container an run it again ? This kind of loops always happen when upgrading and I wasn't able to figure out why.
If the installation did succeeded, the install wizard will be deleted next start you run the container so you won't get into the loop again.

**Comment by sfn84** [comment]
Problem solved...

I used to try restarting the container several times but it was still stuck at Step 7 "Setting up basic configuration".

I have had a look at the script /root/start.sh and noticed it did some checks in **/etc/centreon/** folder which **did not exist**. (Btw, is that normal?)

After creating this folder each steps of the install went smooth.
I just had to restart the container to make it work with the workaround.

I am now able to use centreon 👍
